src/pages/en/guides/migration-guide/from-gatsby.md

```diff
@@ -0,0 +1,1186 @@
+
+```astro title="src/components/Link.astro"
+---
+const { to } = Astro.props
+---
+<a href={to}><slot /></a>
+```
+
+### Gatsby Imports to Astro
+
+Astro requires file imports to reference relative file paths exactly. This can be done using [import aliases](/en/guides/typescript/#import-aliases), or by writing out a relative path in full (e.g. `../../layouts/Layout.astro`). Note that `.astro` and several other file types must be imported with their full file extension.
+
+### Gatsby Children to Astro
+
+Convert any instances of `{children}` to an Astro `<slot />`. Astro does not need to receive `{children}` as a function prop and will automatically render child content in a `<slot />`.
+
+React components that pass multiple sets of children can be migrated to an Astro component using [named slots](https://docs.astro.build/en/core-concepts/astro-components/#named-slots):
+
+```jsx title="pageheader.jsx"
+export const PageHeader = ({navItems, children}) => {
+	return (
+		<header>
+			{children}
+			<nav>
+				<ul>
+					<li><Link to="/">Home</a></li>
+					{navItems}
+				</ul>
+			</nav>
+		</header>
+	)
+}
+```
+
+```jsx title="app.jsx"
+import {PageHeader} from './pageheader';
+
+export const App = () => {
+	return (
+		// ...
+		<PageHeader navItems={<li><Link to="/blog">Blog</a></li>}>
+			<Link to="/">CompanyCo</a>
+		</PageHeader>
+		// ...
+	)
+}
+```
+
+The above component can be reproduced in Astro by using a `slot=""` attribute to pass data to a particular named `<slot />` placeholder component.
+
+```astro title="src/components/PageHeader.astro"
+<header>
+    <nav>
+        <ul>
+            <li><a href="/">Home</a></li>
+            <slot name="navItems"/>
+        </ul>
+    </nav>
+	<slot/>
+</header>
+```
+
+```astro title="src/components/Component.astro"
+---
+import PageHeader from './PageHeader.astro';
+---
+
+<PageHeader>
+	<a href="/">CompanyCo</a>
+	<li slot="navItems"><a href="/blog">Blog</a></li>
+</PageHeader>
+
+<!-- ... -->
+```
+See more about [specific `<slot />` usage in Astro](/en/core-concepts/astro-components/#slots) for details.
+
+### Gatsby Component Props to Astro
+
+To access specific attributes passed to your component (e.g. `<Layout title="About Me"/>`), use `Astro.props`.
+
+The following Gatsby React component:
+
+```jsx title="layout.jsx"
+const Layout = ({title}) => {
+    // ...
+}
+```
+
+Is turned into the following Astro component:
+
+```astro title="Layout.astro"
+---
+const {title} = Astro.props;
+---
+
+<!-- ... -->
+```
+
+#### Type-checking Props
+
+If you're coming from a TypeScript codebase, your Gatsby component might have an interface associated with the component's props:
+
+```jsx title="layout.jsx"
+interface LayoutProps {
+    title: string;
+}
+
+const Layout = ({title}: LayoutProps) => {
+    // ...
+}
+```
+
+In Astro, we can reuse the TypeScript interface and type cast our `Astro.props` destructure to enforce type safety:
```

Comment:
```suggestion
In Astro, reuse the TypeScript interface and type cast the `Astro.props` destructure to enforce type safety:
```
Note to us all, when an edit like this "just works", let's just go ahead and do it!